loopy/__init__.py

```diff
@@ -487,6 +488,68 @@ def make_copy_kernel(new_dim_tags, old_dim_tags=None):
 # }}}
 
 
+# {{{ einsum
+
+def make_einsum(spec, arg_names, **knl_creation_kwargs):
+    """Returns a :class:`LoopKernel` for evaluating array-based
+    operations using Einstein summation convention. Any
+    keyword arguments for kernel creation will be carried through
+    to :func:`make_kernel`.
+
+    :param spec: a string denoting the subscripts for
+        summation as a comma-separated list of subscript labels.
+        This follows the usual :func:`numpy.einsum` convention.
+        Note that the explicit indicator `->` for the precise output
+        form is required.
+    :param arg_names: an iterable of string types denoting
```

Comment:
That's incorrect. `arg_names` is iterated more than once, so it must be a sequence.

Reply:
Done in https://github.com/inducer/loopy/pull/328/commits/5abfc5dbc43024fd139fa3ef404bcb7a1ef569a2